HLTrigger/JetMET/plugins/HLTScoutingPFProducer.cc

```diff
@@ -0,0 +1,249 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/METReco/interface/METCollection.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/Scouting/interface/ScoutingPFJet.h"
+#include "DataFormats/Scouting/interface/ScoutingParticle.h"
+#include "DataFormats/Scouting/interface/ScoutingVertex.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+
+class HLTScoutingPFProducer : public edm::stream::EDProducer<> {
+    public:
+        explicit HLTScoutingPFProducer(const edm::ParameterSet&);
+        ~HLTScoutingPFProducer();
+
+        static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    private:
+        virtual void produce(edm::Event&, const edm::EventSetup&) override;
+
+        edm::EDGetTokenT<reco::PFJetCollection> pfJetCollection_;
+        edm::EDGetTokenT<reco::JetTagCollection> pfJetTagCollection_;
+        edm::EDGetTokenT<reco::PFCandidateCollection> pfCandidateCollection_;
+        edm::EDGetTokenT<reco::VertexCollection> vertexCollection_;
+        edm::EDGetTokenT<reco::METCollection> metCollection_;
+        edm::EDGetTokenT<double> rho_;
+
+        double pfJetPtCut;
+        double pfJetEtaCut;
+        double pfCandidatePtCut;
+
+        bool doJetTags;
+        bool doCandidates;
+        bool doMet;
+};
+
+//
+// constructors and destructor
+//
+HLTScoutingPFProducer::HLTScoutingPFProducer(const edm::ParameterSet& iConfig):
+    pfJetCollection_(consumes<reco::PFJetCollection>(iConfig.getParameter<edm::InputTag>("pfJetCollection"))),
+    pfJetTagCollection_(consumes<reco::JetTagCollection>(iConfig.getParameter<edm::InputTag>("pfJetTagCollection"))),
+    pfCandidateCollection_(consumes<reco::PFCandidateCollection>(iConfig.getParameter<edm::InputTag>("pfCandidateCollection"))),
+    vertexCollection_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vertexCollection"))),
+    metCollection_(consumes<reco::METCollection>(iConfig.getParameter<edm::InputTag>("metCollection"))),
+    rho_(consumes<double>(iConfig.getParameter<edm::InputTag>("rho"))),
+    pfJetPtCut(iConfig.getParameter<double>("pfJetPtCut")),
+    pfJetEtaCut(iConfig.getParameter<double>("pfJetEtaCut")),
+    pfCandidatePtCut(iConfig.getParameter<double>("pfCandidatePtCut")),
+    doJetTags(iConfig.getParameter<bool>("doJetTags")),
+    doCandidates(iConfig.getParameter<bool>("doCandidates")),
+    doMet(iConfig.getParameter<bool>("doMet"))
+{
+    //register products
+    produces<ScoutingPFJetCollection>("scoutingPFJets");
```

Comment:
just a remark: you could put these three collections into the event without an additional label